node/src/accountant/test_utils.rs

```diff
@@ -1644,3 +1678,131 @@ impl ScanSchedulers {
+    estimated_transaction_fee_total_results: Option<u128>,
+    arbitrary_id_stamp_opt: Option<ArbitraryIdStamp>,
+}
+
+impl PayablePaymentsAgent for PayablePaymentsAgentMock {
+    fn conclude_required_fee_per_computed_unit(
+        &mut self,
+        persistent_config: &dyn PersistentConfiguration,
+    ) -> Result<(), PersistentConfigError> {
+        self.conclude_required_fee_per_computed_unit_params
+            .lock()
+            .unwrap()
+            .push(persistent_config.arbitrary_id_stamp());
+        self.conclude_required_fee_per_computed_unit_results
+            .borrow_mut()
+            .remove(0)
+    }
+
+    fn set_up_pending_transaction_id(&mut self, id: U256) {
+        self.set_up_pending_transaction_id_params
+            .lock()
+            .unwrap()
+            .push(id);
+    }
+
+    fn set_up_consuming_wallet_balances(&mut self, balances: ConsumingWalletBalances) {
+        self.set_up_consuming_wallet_balances_params
+            .lock()
+            .unwrap()
+            .push(balances)
+    }
+
+    fn estimated_transaction_fee_total(&self, _number_of_transactions: usize) -> u128 {
+        todo!("to be implemented by GH-711")
+    }
+
+    fn consuming_wallet_balances(&self) -> Option<ConsumingWalletBalances> {
+        todo!("to be implemented by GH-711")
+    }
+
+    fn required_fee_per_computed_unit(&self) -> Option<u64> {
+        self.required_fee_per_computed_unit_results
+            .borrow_mut()
+            .remove(0)
+    }
+
+    fn pending_transaction_id(&self) -> Option<U256> {
+        self.pending_transaction_id_results.borrow_mut().remove(0)
+    }
+
+    arbitrary_id_stamp_in_trait_impl!();
+}
+
+impl PayablePaymentsAgentMock {
+    pub fn conclude_required_fee_per_computed_unit_params(
+        mut self,
+        params: &Arc<Mutex<Vec<ArbitraryIdStamp>>>,
+    ) -> Self {
+        self.conclude_required_fee_per_computed_unit_params = params.clone();
+        self
+    }
+
+    pub fn conclude_required_fee_per_computed_unit_result(
+        self,
+        result: Result<(), PersistentConfigError>,
+    ) -> Self {
+        self.conclude_required_fee_per_computed_unit_results
+            .borrow_mut()
+            .push(result);
+        self
+    }
+
+    pub fn set_up_pending_transaction_id_params(mut self, params: &Arc<Mutex<Vec<U256>>>) -> Self {
+        self.set_up_pending_transaction_id_params = params.clone();
+        self
+    }
+
+    pub fn set_up_consuming_wallet_balances_params(
+        mut self,
+        params: &Arc<Mutex<Vec<ConsumingWalletBalances>>>,
+    ) -> Self {
+        self.set_up_consuming_wallet_balances_params = params.clone();
+        self
+    }
+
+    pub fn estimated_transaction_fee_params(self, _params: &Arc<Mutex<Vec<usize>>>) -> Self {
+        todo!("to be implemented by GH-711")
+    }
+
+    pub fn estimated_transaction_fee_total_result(mut self, result: u128) -> Self {
+        self.estimated_transaction_fee_total_results = Some(result);
```

Comment:
Probably this won't ever be called, because the method in the trait implementation panics.